### PR TITLE
Add delete to story map

### PR DESCRIPTION
**Pull request by IhsenBouallegue**
This PR:
- adds edit to levels
- adds delete dropzone to delete cards
- fixes animation when dragging over incompatible list
- adds zoom (this is a bit problematic due to react beautiful dnd breaking when the zoom changes, but for now it can be used to view the board)
- adds delete to level and case
- refactors story map view

### CODE REVIEW

File: project-canvas/src/components/StoryMapView/Components/Zoom.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <Group>
+        <ActionIcon
+          variant="outline"
+          color="dark"
+          size="sm"
+          onClick={() => setZoomValue(zoomValue - 0.1)}
+        >
+          <IconMinus />
+        </ActionIcon>
+        <Text w={36} align="center">
```
Comment (oussema18) [inline project-canvas/src/components/StoryMapView/Components/Zoom.tsx:22]: using fix value of Pixel is not recommended

File: project-canvas/src/components/StoryMapView/helpers/draggingHelpers.ts [hunk 2]
```diff
@@ -25,8 +25,21 @@ const move = <T>(
   return { newSource: sourceClone, newDestination: destClone }
 }
 
-const isActionList = (caseId: string) => caseId.match(/^a/g)
-const isSubAction = (actionId: string) => actionId.match(/^s/g)
+const remove = <T>(list: T[], index: number) => {
+  const result = Array.from(list)
+  result.splice(index, 1)
```
Comment (oussema18) [inline project-canvas/src/components/StoryMapView/helpers/draggingHelpers.ts:30]: result is not corresponding to its function. we could rename it to something else

**Review by oussema18 — CHANGES_REQUESTED** [review]
I have checked : 
- the name of variables conventions
- if the name expresses its function
- duplicate code
i will trust with the logic if it works 👍 

-

File: project-canvas/src/components/StoryMapView/Components/DeleteDropzone.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Affix, Center, ThemeIcon, Group } from "@mantine/core"
+import { IconTrash } from "@tabler/icons"
+import { StrictModeDroppable } from "../../common/StrictModeDroppable"
+
+export function DeleteDropzone() {
```
[the remark this answers is not in the document]
Comment (IhsenBouallegue) [reply project-canvas/src/components/StoryMapView/Components/DeleteDropzone.tsx:5]: That is on purpose. Due to limitations from react beautiful dnd, I couldn't hide it when not needed. This is only a visual problem as the button can still be clicked.
So let's leave it at that and discuss how we want to solve next time

File: project-canvas/src/components/StoryMapView/helpers/draggingHelpers.ts (hunk 2 above)
Comment (IhsenBouallegue) [reply project-canvas/src/components/StoryMapView/helpers/draggingHelpers.ts:30]: line 29 coppies the input list and then line 30 changes the copy which is the result array. This is a generic function so better keep the names generic.

File: project-canvas/src/components/StoryMapView/helpers/updaterFunctions.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+// Case
+
+export const addCaseFn = (setCases: Updater<Case[]>) => (caseColumn: Case) => {
+  setCases((draft) => {
+    draft.push(caseColumn)
+  })
+}
+
+export const deleteCaseFn = (setCases: Updater<Case[]>) => (caseId: string) => {
+  setCases((draft) => {
```
[the remark this answers is not in the document]
Comment (IhsenBouallegue) [reply project-canvas/src/components/StoryMapView/helpers/updaterFunctions.ts:14]: fixed

**Pull request merged**